src/test/java/com/example/javawebcourse/api/web/CosmoCatControllerTest.java

```diff
@@ -0,0 +1,37 @@
+package com.example.javawebcourse.api.web;
+
+import com.example.javawebcourse.web.CosmoCatController;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import com.example.javawebcourse.api.annotation.DisabledFeatureToggle;
+import com.example.javawebcourse.api.annotation.EnabledFeatureToggle;
+import com.example.javawebcourse.featuretoggle.FeatureToggles;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(CosmoCatController.class)
+class CosmoCatControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    @DisabledFeatureToggle(FeatureToggles.COSMO_CATS)
+    void shouldGet404FeatureDisabled() throws Exception {
+        mockMvc.perform(get("/api/cosmo-cats")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
```

Comment:
```suggestion
    @Test
    @DisplayName("Verify returns 404 status code if feature is desabled")
    @DisabledFeatureToggle(FeatureToggles.COSMO_CATS)
    void testFeatureDiabled() throws Exception {
        mockMvc.perform(get("/api/cosmo-cats")
                .andExpect(status().isNotFound());
```

It is genrally good idea to keep test case description in @DisplayName or similar. It makes it more comfortable to look on test reoirts in IDE, and also saves you from 50-meter long method names. Also contentType is redundant